File: facebookCodingPuzzles/Level1/kaitenzushi/KaitenzushiTest.cs
namespace Practice.FacebookCodingPuzzles.Level1
{
    using NUnit.Framework;

    public class KaitenzushiTest
    {
        [Test]
        public void Example1()
        {
            // Arrange
            var N = 6;
            var D = new int[] {1, 2, 3, 3, 2, 1};
            var K = 1;

            var expected = 5;

            // Act
            var actual = Kaitenzushi.GetMaximumEatenDishCount(N, D, K);

            // Assert
            Assert.AreEqual(expected, actual);
        }

        [Test]
        public void Example2()
        {
            // Arrange
            var N = 6;
            var D = new int[] {1, 2, 3, 3, 2, 1};
            var K = 2;

            var expected = 4;

            // Act
            var actual = Kaitenzushi.GetMaximumEatenDishCount(N, D, K);

            // Assert
            Assert.AreEqual(expected, actual);
        }

        [Test]
        public void Example3()
        {
            // Arrange
            var N = 7;
            var D = new int[] {1, 2, 1, 2, 1, 2, 1};
            var K = 2;

            var expected = 4;

            // Act
            var actual = Kaitenzushi.GetMaximumEatenDishCount(N, D, K);

            // Assert
            Assert.AreEqual(expected, actual);
        }
    }
}

File: facebookCodingPuzzles/level1/kaitenzushi/kaitenzushi.test.js
"use strict";

const getMaximumEatenDishCount = require("./kaitenzushi.js");

test("Example 1 returns 5", () => {
    expect(getMaximumEatenDishCount(6, [1, 2, 3, 3, 2, 1], 1)).toBe(5);
});

test("Example 2 returns 5", () => {
    expect(getMaximumEatenDishCount(6, [1, 2, 3, 3, 2, 1], 2)).toBe(4);
});

test("Example 3 returns 5", () => {
    expect(getMaximumEatenDishCount(7, [1, 2, 1, 2, 1, 2, 1], 2)).toBe(2);
});